lib/kitchen/transport/ssh.rb

```diff
@@ -226,6 +235,49 @@ def wait_until_ready
         # @api private
         attr_reader :port
 
+        # @return [String] The ssh gateway to use when connecting to the
+        #   remote SSH host
+        # @api private
+        attr_reader :ssh_gateway
+
+        # @return [String] The username to use when using an ssh gateway
+        # @api private
+        attr_reader :ssh_gateway_username
+
+        # Establish an SSH session on the remote host using a gateway host.
+        #
+        # @param opts [Hash] retry options
+        # @option opts [Integer] :retries the number of times to retry before
+        #   failing
+        # @option opts [Float] :delay the number of seconds to wait until
+        #   attempting a retry
+        # @option opts [String] :message an optional message to be logged on
+        #   debug (overriding the default) when a rescuable exception is raised
+        # @return [Net::SSH::Connection::Session] the SSH connection session
+        # @api private
+        def establish_connection_via_gateway(opts)
+          logger.debug("[SSH] opening connection to #{self} via " \
+            "#{ssh_gateway_username}@#{ssh_gateway}")
+          Net::SSH::Gateway.new(ssh_gateway,
+            ssh_gateway_username, options).ssh(hostname, username, options)
+        rescue *RESCUE_EXCEPTIONS_ON_ESTABLISH => e
```

Comment:
I tried. Look at this commit https://github.com/test-kitchen/test-kitchen/pull/1091/commits/d171410a2dc1e7d8a2b34c0df84168069dfefb33. But rubocop was very unhappy -(